client/client.go

```diff
@@ -21,20 +21,28 @@ type Client struct {
 	local  *tuf.TufRepo
 	remote store.RemoteStore
 	keysDB *keys.KeyDB
+	cache  *store.FileCacheStore
 }
 
-func NewClient(local *tuf.TufRepo, remote store.RemoteStore, keysDB *keys.KeyDB) *Client {
+func NewClient(local *tuf.TufRepo, remote store.RemoteStore, keysDB *keys.KeyDB, cachePath string) *Client {
+	cache := store.NewFileCacheStore(cachePath)
 	return &Client{
 		local:  local,
 		remote: remote,
 		keysDB: keysDB,
+		cache:  cache,
 	}
 }
 
-// Update an in memory copy of the TUF Repo. If an error is returned, the
-// Client instance should be considered corrupted and discarded as it may
-// be left in a partially updated state
 func (c *Client) Update() error {
+	// 1. Get timestamp
+	//   a. If timestamp error (verification, expired, etc...) download new root and return to 1.
+	// 2. Check if local snapshot is up to date
+	//   a. If out of data, get updated snapshot
+	//     i. If snapshot error, download new root and return to 1.
+	// 3. Check if root correct against snapshot
```

Comment:
lowercase `update()` will never be called more than twice (as specified in the TUF spec, and in that update flow diagram I put together).